### PR TITLE
Use joinsession if session id is used

### PR DESCRIPTION
**Pull request by dominikl**
Bugfix for https://github.com/ome/omero-insight/issues/293 .

### CODE REVIEW

**Comment by jburel** [comment]
Screenshot 2022-05-06 at 09 25 07
insight connected using session key

**Comment by dominikl** [comment]
👍 I have to add this case to the integration test too.

**Comment by jburel** [comment]
It works the first time but not the second time
``SecurityViolation: 1 methods active.``

**Comment by dominikl** [comment]
Integration test: https://github.com/ome/openmicroscopy/pull/6319

**Comment by dominikl** [comment]
Thanks @jburel . I think that should be all cases covered now. Should now somehow be tested together with the integration test.

**Comment by dominikl** [comment]
Taking the discussions https://github.com/ome/openmicroscopy/pull/6319 into account:
Instead of the proposed changes, shall I rather remove all calls to `joinSession()` and use only `createSession()`? This means all checks if a username is a session id can be removed too. But also implies, that if you want to login with a session id, you have to provide it as username and password. At the moment you only have to provide it as username (same for Insight).

File: src/main/java/omero/gateway/Gateway.java [hunk 1]
```diff
@@ -366,69 +366,6 @@ public void disconnect() {
             cacheService.shutDown();
     }
 
-    /**
```
Comment (jburel) [inline src/main/java/omero/gateway/Gateway.java:369]: insight has a method ``joinSession``. It does not seem to be used but some changes will be required there too

Comment (jburel) [reply src/main/java/omero/gateway/Gateway.java:369]: I forgot about that

**Pull request closed without merging**